presto-spark/src/main/java/com/facebook/presto/spark/execution/PrestoSparkTaskCompiler.java

```diff
@@ -0,0 +1,340 @@
+import com.facebook.presto.metadata.SessionPropertyManager;
+import com.facebook.presto.operator.Driver;
+import com.facebook.presto.operator.DriverContext;
+import com.facebook.presto.operator.DriverFactory;
+import com.facebook.presto.operator.TaskContext;
+import com.facebook.presto.operator.TaskStats;
+import com.facebook.presto.spark.SparkTaskDescriptor;
+import com.facebook.presto.spark.classloader_interface.IPrestoSparkTaskCompiler;
+import com.facebook.presto.spi.block.BlockEncodingSerde;
+import com.facebook.presto.spi.memory.MemoryPoolId;
+import com.facebook.presto.spi.plan.PlanNodeId;
+import com.facebook.presto.spiller.NodeSpillConfig;
+import com.facebook.presto.spiller.SpillSpaceTracker;
+import com.facebook.presto.sql.planner.LocalExecutionPlanner;
+import com.facebook.presto.sql.planner.LocalExecutionPlanner.LocalExecutionPlan;
+import com.facebook.presto.sql.planner.PlanFragment;
+import com.google.common.collect.AbstractIterator;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterators;
+import io.airlift.slice.DynamicSliceOutput;
+import io.airlift.slice.SliceOutput;
+import io.airlift.slice.Slices;
+import io.airlift.units.DataSize;
+import org.apache.spark.util.CollectionAccumulator;
+import scala.Tuple2;
+
+import javax.inject.Inject;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.OptionalInt;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static com.facebook.presto.execution.buffer.PagesSerdeUtil.readSerializedPages;
+import static com.facebook.presto.execution.buffer.PagesSerdeUtil.writeSerializedPage;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static java.util.Objects.requireNonNull;
+
+public class PrestoSparkTaskCompiler
+        implements IPrestoSparkTaskCompiler
+{
+    private static final Logger log = Logger.get(PrestoSparkTaskCompiler.class);
+
+    private final SessionPropertyManager sessionPropertyManager;
+
+    private final JsonCodec<SparkTaskDescriptor> sparkTaskDescriptorJsonCodec;
+    private final JsonCodec<TaskStats> taskStatsJsonCodec;
+
+    private final Executor notificationExecutor;
+    private final ScheduledExecutorService yieldExecutor;
+
+    private final LocalExecutionPlanner localExecutionPlanner;
+    private final BlockEncodingSerde blockEncodingSerde;
+
+    private final DataSize maxUserMemory;
+    private final DataSize maxTotalMemory;
+    private final DataSize maxSpillMemory;
+
+    private final boolean perOperatorCpuTimerEnabled;
+    private final boolean cpuTimerEnabled;
+
+    @Inject
+    public PrestoSparkTaskCompiler(
```

Comment:
Note this class did the following:
- Compile fragmented plan to operator chain (a.k.a `DriverFactory`)
- Associate split to `DriverFactory` to create drivers
- Returns a `SparkSqlExecutor` that executes these drivers (for now it's simple round-robin, in the future we might want to handle multiple threading)

So "compiling" is just the first step. What about `SparkSqlExecutorFactory`?

Reply:
Renamed to `PrestoSparkExecutorFactory` to be consistent